applications/tari_console_wallet/src/ui/state/wallet_event_monitor.rs

```diff
@@ -79,22 +85,37 @@ impl WalletEventMonitor {
                         match result {
                             Ok(msg) => {
                                 trace!(target: LOG_TARGET, "Wallet Event Monitor received wallet event {:?}", msg);
-                                     match (*msg).clone() {
-                                        ConnectivityEvent::PeerDisconnected(_) |
-                                        ConnectivityEvent::ManagedPeerDisconnected(_) |
-                                        ConnectivityEvent::PeerConnected(_) |
-                                        ConnectivityEvent::PeerBanned(_) |
-                                        ConnectivityEvent::PeerOffline(_) |
-                                        ConnectivityEvent::PeerConnectionWillClose(_, _) => {
-                                            self.trigger_peer_state_refresh().await;
-                                        },
-                                        /// Only the above variants trigger state refresh
-                                        _ => (),
-                                     }
+                                match (*msg).clone() {
+                                    ConnectivityEvent::PeerDisconnected(_) |
+                                    ConnectivityEvent::ManagedPeerDisconnected(_) |
+                                    ConnectivityEvent::PeerConnected(_) |
+                                    ConnectivityEvent::PeerBanned(_) |
+                                    ConnectivityEvent::PeerOffline(_) |
+                                    ConnectivityEvent::PeerConnectionWillClose(_, _) => {
+                                    self.trigger_peer_state_refresh().await;
+                                    },
+                                    /// Only the above variants trigger state refresh
+                                    _ => (),
+                                }
                             },
                             Err(_) => debug!(target: LOG_TARGET, "Lagging read on Connectivity event broadcast channel"),
                         }
                     },
+                    result = base_node_events.select_next_some() => {
+                        match result {
+                            Ok(msg) => {
+                                trace!(target: LOG_TARGET, "Wallet Event Monitor received base node event {:?}", msg);
+                                match (*msg).clone() {
+                                    BaseNodeEvent::BaseNodeState(state) => {
+                                        debug!(target: LOG_TARGET, "new base node state: {:?}", state);
+                                        self.trigger_base_node_state_refresh(state).await;
+                                    },
+                                    _ => warn!(target: LOG_TARGET, "Received an unhandled BaseNodeEvent"),
```

Comment:
I don't think you log this, especially as a warn. This monitor just doesn't care about the other variants, its not an error if it receives them. See the comment in the match above for the variants of the ConnectivityEvent we don't care about.

Reply:
In fact is this branch even possible on this enum? If this wasn't in the select! macro I believe clippy would have something to say.

Reply:
Gotcha will do